packages/shared/lib/services/proxy/retry.ts

```diff
@@ -68,22 +68,38 @@ export function getProxyRetryFromErr({ err, proxyConfig }: { err: unknown; proxy
     if (proxyConfig.retryHeader && (proxyConfig.retryHeader.at || proxyConfig.retryHeader.after)) {
         // Headers configured on the fly
         const type = proxyConfig.retryHeader.at ? 'at' : 'after';
-        const retryHeader = proxyConfig.retryHeader.at ? proxyConfig.retryHeader.at : proxyConfig.retryHeader.after;
+        let retryHeaders: string[];
 
-        const res = getRetryFromHeader({ err, type, retryHeader: retryHeader! });
-        if (res.found) {
-            return { retry: true, reason: `custom_${res.reason}`, wait: res.wait };
+        if (proxyConfig.retryHeader.at) {
+            retryHeaders = Array.isArray(proxyConfig.retryHeader.at) ? proxyConfig.retryHeader.at : [proxyConfig.retryHeader.at];
+        } else {
+            retryHeaders = [proxyConfig.retryHeader.after!];
+        }
+
+        for (const retryHeader of retryHeaders) {
+            const res = getRetryFromHeader({ err, type, retryHeader });
+            if (res.found) {
+                return { retry: true, reason: `custom_${res.reason}`, wait: res.wait };
+            }
         }
     }
```

Comment:
what if the response has the multiple specified headers with different values? Should we take into account the one with the most restrictive retry time instead of the first one found?